ovs: Raise NmstateValueError when desired interface is orphan

cathay4t opened the pull request:
When orphan interface exists in desire state, raise NmstateValueError.

When orphan interface exists in desire state, but caused by absent of
existing parent, mark this orphan interface as absent.

Integration test case added.

ffmancera commented on libnmstate/ifaces/ifaces.py:
```diff
@@ -380,9 +380,15 @@ def _mark_orphan_as_absent(self):
+                ):
                     iface.mark_as_changed()
                     iface.state = InterfaceState.ABSENT
+                elif parent_iface is None and iface.is_desired:
+                    raise NmstateValueError(
+                        f"Failed to find parent interface for {iface.name}"
+                    )
```
The integration test added in this commit is covering this case. I don't know why codecov complains about this line :-(

ffmancera reviewed (approved):
Thanks!

cathay4t committed:
ovs: Raise NmstateValueError when desired interface is orphan

When orphan interface exists in desire state, raise NmstateValueError.

When orphan interface exists in desire state, but caused by absent of
existing parent, mark this orphan interface as absent.

Integration test case added.

Signed-off-by: Gris Ge <fge@redhat.com>
